src/instruments/src/Clock/Components/Clock.tsx

```diff
@@ -12,33 +11,85 @@ const secondsToDisplay = (seconds: number): number[] => {
     return displayTime;
 };
 
-export const Clock = () => {
-    const [ltsTest] = useSimVar('L:A32NX_OVHD_INTLT_ANN', 'bool', 250);
-    const [dateMode, setDateMode] = useState(false);
-    const [currentUTC] = useSimVar('E:ZULU TIME', 'seconds', 200);
-    const [dayOfMonth] = useSimVar('E:ZULU DAY OF MONTH', 'number', 1000);
-    const [monthOfYear] = useSimVar('E:ZULU MONTH OF YEAR', 'number', 1000);
-    const [year] = useSimVar('E:ZULU YEAR', 'number', 1000);
-
-    useInteractionEvent('A32NX_CHRONO_DATE', () => {
-        setDateMode(!dateMode);
-    });
-
-    let text1;
-    let text2;
-    if (!dateMode) {
-        const displayTime = secondsToDisplay(currentUTC);
-        text1 = `${displayTime[0].toString().padStart(2, '0')}:${displayTime[1].toString().padStart(2, '0')}`;
-        text2 = displayTime[2].toString().padStart(2, '0');
-    } else {
-        text1 = `${monthOfYear.toString().padStart(2, '0')}:${dayOfMonth.toString().padStart(2, '0')}`;
-        text2 = year.toString().substr(2, 4);
+interface ClockProps extends ComponentProps {
+    bus: EventBus;
+}
+
+export class Clock extends DisplayComponent<ClockProps> {
+    private readonly clockTextBig = Subject.create('');
+
+    private readonly clockTextSmall = Subject.create('');
+
+    private readonly ltsTest = Subject.create(false);
+
+    private readonly dcHot1IsPowered = Subject.create(true);
+
+    private readonly dateMode = Subject.create(false);
+
+    private readonly currentUTC = Subject.create(0);
+
+    private readonly currentDate = Subject.create({ dayOfMonth: 1, monthOfYear: 1, year: 1970 });
+
+    private updateClockText(): void {
+        if (!this.dcHot1IsPowered.get()) {
+            this.clockTextBig.set('');
+            this.clockTextSmall.set('');
+            return;
+        }
+
+        if (this.ltsTest.get()) {
+            this.clockTextBig.set('88:88');
+            this.clockTextSmall.set('88');
+            return;
+        }
+
+        if (!this.dateMode.get()) {
+            const displayTime = secondsToDisplay(this.currentUTC.get());
+            this.clockTextBig.set(`${displayTime[0].toString().padStart(2, '0')}:${displayTime[1].toString().padStart(2, '0')}`);
+            this.clockTextSmall.set(displayTime[2].toString().padStart(2, '0'));
```

Comment:
perhaps use enum values instead of unlabelled magic indices for clarity

Reply:
Done 👍